### PR TITLE
Make the source compatible to java 1.8

### PR DESCRIPTION
**Pull request by fabiojose**
Signed-off-by: Fabio José <fabiojose@gmail.com>

In our company we have strong compliance rules and at this moment we are able to use just java 1.8.

Could you accept these changes to make the impl. compatible to java 1.8?

### CODE REVIEW

File: api/src/test/java/io/cloudevents/CloudEventBuilderTest.java [hunk 1]
```diff
@@ -30,15 +31,18 @@
...
         final String eventId = UUID.randomUUID().toString();
         final URI src = URI.create("/trigger");
         final String eventType = "My.Cloud.Event.Type";
         final String eventTypeVersion = "2.0";
         final ZonedDateTime eventTime = ZonedDateTime.now();
         final String contentType = "application/json";
         final URI schemaUri = URI.create("http://cloudevents.io/schema");
-        final Map<String, String> extensionData = Map.of("foo", "bar");
+        final Map<String, String> extensionData = new HashMap<>();
+        extensionData.put("foo", "bar");
```
Comment (matzew) [inline api/src/test/java/io/cloudevents/CloudEventBuilderTest.java]: :crying_cat_face:   but agreed :smile:

File: api/src/main/java/io/cloudevents/impl/ZonedDateTimeSerializer.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+public class ZonedDateTimeSerializer extends StdSerializer<ZonedDateTime> {
```
Comment (matzew) [inline api/src/main/java/io/cloudevents/impl/ZonedDateTimeSerializer.java]: thanks!

**Comment by matzew** [comment]
@fabiojose thanks for the PR and the update!

File: api/src/test/java/io/cloudevents/CloudEventBuilderTest.java (hunk 1 above)
Comment (fabiojose) [reply api/src/test/java/io/cloudevents/CloudEventBuilderTest.java]: The java9 solution to construct maps is amazing!

Any idea to release it?

File: api/src/main/java/io/cloudevents/impl/ZonedDateTimeSerializer.java (hunk 2 above)
Comment (fabiojose) [reply api/src/main/java/io/cloudevents/impl/ZonedDateTimeSerializer.java]: You are welcome!

**Pull request merged**